rskj-core/src/main/java/co/rsk/net/BlockStore.java

```diff
@@ -144,6 +146,24 @@ public synchronized Block getBlockByHash(byte[] hash) {
         return ls;
     }
 
+    /**
+     * getChildrenOf returns all the children of a list of blocks that are in the BlockStore.
+     *
+     * @param blocks the list of blocks to retrieve the children.
+     * @return A list with all the children of the given list of blocks.
+     */
+    public List<Block> getChildrenOf(List<Block> blocks) {
+        final List<Block> children = new ArrayList<>();
+
+        // TODO(mc) check if we need a better implementation because BlockUtils.addBlocksToList is O(N^2)
```

Comment:
Block.getHash -> WrapByteArray -> HashSet -> getBlockByParents -> Add all to a list, there will be no repeated blocks

Reply:
Changed implementation to the one suggested